### PR TITLE
Assignment(JavaScript): Completed tasks of Iterator,Generator,Closures,ES5 and ES6 

### PR DESCRIPTION
**Pull request by DeqodeShubhamKaushalya**
- Worked on ES5 feature
      Callback
      Error handling using callback
- Worked on ES6 feature
      Error handling using try catch
-  Generator: To generate generic counter
-  Use of Iterator to make object compatible with for...of loop
-  Closures

### CODE REVIEW

File: ES5-CRUD-Operation.js [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+function StudentDTO(id, name) {
+    this.id = id;
+    this.name = name;
+}
+
+function StudentDAO() {
+
+}
+
+StudentDAO.prototype.add = (StudentDTO, callback) => {
```
Comment (abparmarSK) [inline ES5-CRUD-Operation.js:25]: s should be in lower case for StudentDTO

File: ES5-CRUD-Operation.js [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    if (Number.isInteger(StudentDTO.id)) {
```
Comment (abparmarSK) [inline ES5-CRUD-Operation.js:26]: s should be in lower case for StudentDTO

File: ES5-CRUD-Operation.js [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+            }
+        } else {
+            callback(new Error('Student name must be a valid string.'));
+        }
+    } else {
+        callback(new Error('Student Id must be a valid number.'));
+    }
+};
+
+StudentDAO.prototype.update = (StudentDTO, callback) => {
```
Comment (abparmarSK) [inline ES5-CRUD-Operation.js:44]: s should be in lower case for StudentDTO

File: ES5-CRUD-Operation.js [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        callback(new Error('Student Id must be a valid number.'))
+    }
+};
+
+StudentDAO.prototype.getAll = (callback) => {
+    let studentList = students.map(student => student.name);
+    callback(null, studentList);
+};
+
+StudentDAO.prototype.getById = (StudentDTO, callback) => {
```
Comment (abparmarSK) [inline ES5-CRUD-Operation.js:67]: s should be in lower case for StudentDTO

File: ES5-CRUD-Operation.js [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+            callback(null,students[getIndexofId]);
+        } else {
+            callback(new Error('Student not found by given ID.'))
+        }
+    } else {
+        callback(new Error('Student Id must be a valid number.'))
+    }
+};
+
+StudentDAO.prototype.delete = (StudentDTO, callback) => {
```
Comment (abparmarSK) [inline ES5-CRUD-Operation.js:80]: s should be in lower case for StudentDTO

File: ES5-CRUD-Operation.js (hunk 3 above)
Comment (DeqodeShubhamKaushalya) [reply ES5-CRUD-Operation.js]: It is changed from student studentDto

File: ES5-CRUD-Operation.js (hunk 4 above)
Comment (DeqodeShubhamKaushalya) [reply ES5-CRUD-Operation.js]: It is changed from student studentDto

File: ES5-CRUD-Operation.js (hunk 5 above)
Comment (DeqodeShubhamKaushalya) [reply ES5-CRUD-Operation.js]: It is changed from student studentDto

File: ES5-CRUD-Operation.js (hunk 1 above)
Comment (DeqodeShubhamKaushalya) [reply ES5-CRUD-Operation.js]: It is changed from student studentDto

File: ES5-CRUD-Operation.js (hunk 2 above)
Comment (DeqodeShubhamKaushalya) [reply ES5-CRUD-Operation.js]: It is changed from student studentDto

**Pull request merged**